public/app/plugins/panel/singlestat2/SingleStatPanel.tsx

```diff
@@ -24,13 +24,31 @@ export const getSingleStatValues = (props: PanelProps<SingleStatBaseOptions>): D
     theme: config.theme,
   });
 
-  return processTimeSeries({
-    data,
-    nullValueMode: NullValueMode.Null,
-  }).map((series, index) => {
-    const value = stat !== 'name' ? series.stats[stat] : series.label;
-    return processor(value);
+  const values: DisplayValue[] = [];
+  data.forEach(t => {
```

Comment:
prefer for of loops for all loops (10x more performant as well for large arrays as it compiles to native for loop)

Reply:
👍 good to know.  In this case that loop is on the list of tables, not the rows.  I changed it to for loop and will push soon